### PR TITLE
Refactor finding IP address for primary interface

### PR DESCRIPTION
**Pull request by jasonzio**

### CODE REVIEW

File: azurelinuxagent/common/osutil/default.py [hunk 1]
```diff
@@ -669,49 +672,71 @@ def get_if_mac(self, ifname):
...
-            else:
-                # use this one
-                logger.info('Interface [{0}] selected'.format(iface))
-                break
+        sock = None
+        ifconf_buff = buff.tostring()
+
+        ifaces = {}
+        for i in range(0, array_size, struct_size):
+            iface = ifconf_buff[i:i+16].split(b'\0', 1)[0]
```
Comment (boumenot) [inline azurelinuxagent/common/osutil/default.py]: 16 should be IFNAMSIZ.

File: azurelinuxagent/common/osutil/default.py [hunk 2]
```diff
@@ -669,49 +672,71 @@ def get_if_mac(self, ifname):
...
+            if len(iface) > 0:
+                iface_name = iface.decode('latin-1')
+                if iface_name not in ifaces:
+                    ifaces[iface_name] = socket.inet_ntoa(ifconf_buff[i+20:i+24])
+        return ifaces
+
+
+    def get_first_if(self):
+        """
+        Return the interface name, and ip addr of the
```
Comment (boumenot) [inline azurelinuxagent/common/osutil/default.py]: NIT: ip -> IPv4

File: azurelinuxagent/common/osutil/default.py (hunk 1 above)
Comment (jasonzio) [reply azurelinuxagent/common/osutil/default.py]: Fixed.

File: azurelinuxagent/common/osutil/default.py (hunk 2 above)
Comment (jasonzio) [reply azurelinuxagent/common/osutil/default.py]: I updated this entire docstring to reflect the actual goal (return the IPv4 of the primary interface; if not available, the IPv4 of any interface).

**Pull request merged**